src/supermarket/app/lib/schema_definer/symbolize_struct.rb

```diff
@@ -0,0 +1,7 @@
+module SchemaDefiner
+  class SymbolizeStruct < Dry::Struct
```

Comment:
Why do we need two modules ? Can this be combined to one module instead ? there is just one method and one class here ..  can we implement this class in types.rb also ?

Reply:
Actually there is need for one module and one class. In the current implementation also there are 2.

The class is `Dry::Struct` but along with that class I wanted to add certain behaviours i.e. `transform_keys(&:to_sym)`. For this reason I created a class as an interface in which you can attach different behaviours to be used by many classes (the implementations) and you can scale it the way you want.

But the module is a specific component of the dry-types gem which needs to be kept separate for need basis. But yes they could be part of the same file.